tools/model-compatibility-tester/test-model-compatibility.sh

```diff
@@ -0,0 +1,489 @@
+    find "$ai_dir" -name "*.md" -type f | sed 's|.*/||; s|\.md$||' | sed 's|^|ai/|' | sort
+}
+
+# Discover repositories from Docker Hub namespace
+discover_namespace_repositories() {
+    local namespace="$1"
+    local page=1
+    local repositories=()
+    local timestamp=$(date '+%Y-%m-%d %H:%M:%S')
+    
+    echo "[$timestamp] [INFO] Discovering repositories from namespace: $namespace" >&2
+    
+    while true; do
+        local url="https://hub.docker.com/v2/namespaces/$namespace/repositories/?page=$page&page_size=100"
+        local response=$(curl -s "$url")
+        
+        # Extract repository names from JSON response
+        local repo_names=$(echo "$response" | grep -o '"name":"[^"]*"' | sed 's/"name":"//g' | sed 's/"//g')
+        
+        if [[ -z "$repo_names" ]]; then
+            break
+        fi
+        
+        repositories+=($repo_names)
+        
+        # Check if there's a next page
+        local next_page=$(echo "$response" | grep -o '"next":"[^"]*"')
+        if [[ -z "$next_page" ]]; then
+            break
+        fi
+        
+        ((page++))
+    done
+    
+    if [[ ${#repositories[@]} -gt 0 ]]; then
+        echo "[$timestamp] [INFO] Found ${#repositories[@]} repositories in namespace $namespace" >&2
+        printf '%s\n' "${repositories[@]}"
+    else
+        echo "[$timestamp] [INFO] Found 0 repositories in namespace $namespace" >&2
+    fi
+}
+
+# Get all tags for a repository (except latest)
+get_repository_tags() {
+    local namespace="$1"
+    local repository="$2"
+    local page=1
+    local tags=()
+    local timestamp=$(date '+%Y-%m-%d %H:%M:%S')
+    
+    echo "[$timestamp] [INFO] Getting tags for $namespace/$repository" >&2
+    
+    while true; do
+        local url="https://hub.docker.com/v2/repositories/$namespace/$repository/tags/?page=$page&page_size=100"
+        local response=$(curl -s "$url")
+        
+        # Extract tag names from JSON response, excluding "latest"
+        local tag_names=$(echo "$response" | grep -o '"name":"[^"]*"' | sed 's/"name":"//g' | sed 's/"//g' | grep -v "^latest$")
+        
+        if [[ -z "$tag_names" ]]; then
+            break
+        fi
+        
+        tags+=($tag_names)
+        
+        # Check if there's a next page
+        local next_page=$(echo "$response" | grep -o '"next":"[^"]*"')
+        if [[ -z "$next_page" ]]; then
+            break
+        fi
+        
+        ((page++))
+    done
+    
+    if [[ ${#tags[@]} -gt 0 ]]; then
+        echo "[$timestamp] [INFO] Found ${#tags[@]} tags for $namespace/$repository (excluding latest)" >&2
+        printf '%s\n' "${tags[@]}"
+    else
+        echo "[$timestamp] [INFO] Found 0 tags for $namespace/$repository (excluding latest)" >&2
+    fi
+}
+
+# Get model variants (fallback for non-namespace mode)
+get_model_variants() {
+    local model="$1"
+    
+    # For now, return common quantization patterns
+    echo -e "latest\n8B-Q4_K_M\n8B-Q8_0\n8B-F16\n7B-Q4_K_M\n7B-Q8_0"
```

Comment:
```suggestion
    crane ls "$model"
```
@ekcasey's suggestion; seems to work well - I think the `crane` dependency is okay for an internal tool.